src/codegen.cpp

```diff
@@ -386,8 +386,11 @@ struct jl_tbaacache_t {
         MDNode *tbaa_mutab_scalar;
         std::tie(tbaa_mutab, tbaa_mutab_scalar) =
             tbaa_make_child(mbuilder, "jtbaa_mutab", tbaa_value_scalar);
+        MDNode *tbaa_immut_scalar;
         tbaa_datatype = tbaa_make_child(mbuilder, "jtbaa_datatype", tbaa_mutab_scalar).first;
-        tbaa_immut = tbaa_make_child(mbuilder, "jtbaa_immut", tbaa_value_scalar).first;
+        tbaa_mutable_unionselbyte = tbaa_make_child(mbuilder, "jtbaa_mut_unionselbyte", tbaa_mutab_scalar).first;
+        std::tie(tbaa_immut, tbaa_immut_scalar) = tbaa_make_child(mbuilder, "jtbaa_immut", tbaa_value_scalar);
+        tbaa_immut_unionselbyte = tbaa_make_child(mbuilder, "jtbaa_immut_unionselbyte", tbaa_immut_scalar).first;
```

Comment:
This expresses that `tbaa_immut_unionselbyte` does not alias `tbaa_stack` - is that what you intended?

Reply:
It doesn't (it never did which is why it was odd that is was there).

Reply:
Is this not a usage on the stack? https://github.com/JuliaLang/julia/blob/4b90899c7dfc98db75ff47e9005d208d69aa29d9/src/cgutils.cpp#L4328

Reply:
But that means it doesn't alias tbaa_immut which causes some issues 😄